src/test_runner/failure_persistence/file.rs

```diff
@@ -144,159 +233,118 @@ where
     }
 }
 
-impl FailurePersistence for FileFailurePersistence {
-    fn load_persisted_failures(&self, source_file: Option<&'static str>) -> Vec<[u32; 4]> {
-        let p = self.resolve(
-                source_file
-                    .and_then(|s| absolutize_source_file(Path::new(s)))
-                    .as_ref()
-                    .map(|cow| &**cow));
-
-        let path: Option<&PathBuf> = match p {
-            Some(ref inner_path) => Some(inner_path),
-            None => None,
-        };
-        let result: io::Result<Vec<[u32; 4]>> = path.map_or_else(
-            || Ok(vec![]),
-            |path| {
-                // .ok() instead of .unwrap() so we don't propagate panics here
-                let _lock = PERSISTENCE_LOCK.read().ok();
+fn parse_seed_line(mut line: String, path: &Path, lineno: usize) -> Option<Seed> {
+    // Remove anything after and including '#':
+    if let Some(comment_start) = line.find('#') {
+        line.truncate(comment_start);
+    }
 
-                let mut ret = Vec::new();
+    // Split by whitespace and ignore empty lines:
+    let parts = line.trim().split(char::is_whitespace).collect::<Vec<_>>();
+    let len = parts.len();
+    if len == 0 { return None; }
+
+    // Decide which parser to use depending on format:
+    let first_ok = parts[0] == "xs";
+    let parser = match len {
+        5 if first_ok => {
```

Comment:
The point of the leading `xs` was to be able to handle things like this, so it would be better to change the type.

Reply:
Not sure I follow what you mean by *"change the type"*...?

Reply:
`xs` is the type of the line. So `xs` would mean "four u32s" and we create a new line type for the new format.

Reply:
(The name is for `XorShift` because I didn't anticipate the type of the seed ever changing.)

Reply:
Oh; now I get it! I didn't realize `xs` stood for `XorShift` at all ;)

Sounds like a good idea to change the prefix 👍